pkg/storages/s3/uploader.go

```diff
@@ -74,8 +77,16 @@ func (uploader *Uploader) createUploadInput(bucket, path string, content io.Read
 }
 
 func (uploader *Uploader) upload(bucket, path string, content io.Reader) error {
+	var err error
+	uploadTimeout := uploader.uploadTimeout
 	input := uploader.createUploadInput(bucket, path, content)
-	_, err := uploader.uploaderAPI.Upload(input)
+	if uploadTimeout != 0 {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Duration(uploadTimeout * int(time.Second)))
+		defer cancel()
+		_, err = uploader.uploaderAPI.UploadWithContext(ctx, input)
```

Comment:
Single timeout to the whole upload is not very useful.
The upload may take 10 minutes or 10 hours in normal situation depending on the file size.

So it will be difficult to choose the one timeout to fit them all.